src/entity/entity.cpp

```diff
@@ -188,6 +188,51 @@ void EntitySystem::remove(EntityID id) {
 }
 
 void EntitySystem::update() {
+#if IMGUI_ENABLE
+    for (auto [id, e] : entities) {
+        Physics::AABody box = {};
+        if (Vec3 *p = get_field_by_name_no_warn<Vec3>(e, FieldName::position)) {
+            box.position = *p;
+        }
+
+        // Size for boxes that doesn't have a scale.
+        box.half_size = Vec3(1.0, 1.0, 1.0) * 0.2;
+        if (Vec3 *s = get_field_by_name_no_warn<Vec3>(e, FieldName::scale)) {
+            box.half_size = (*s) * 0.5;
+        }
+
+        Vec3 start = GFX::current_camera()->position;
+        Vec3 dir = GFX::current_camera()->get_forward();
+        Physics::Manifold manifold = Physics::collision_line_aabody(start, dir, &box);
+        // TODO(ed): Only select the closest.
+        bool hovering = manifold.t > 0;
+
+        Vec4 nothing_color = { 0., 0., 0., 1. };
+        Vec4 hovering_color = { 0., 0.5, 0.5, 1. };
+        Vec4 selected_color = { 0., 0.5, 0., 1. };
+        Vec4 selected_and_hover_color = { 0., 0.5, 0.25, 1. };
```

Comment:
Maybe my color vision is bad or something but I don't notice a difference between this color and `selected_color`. What about (0, 0.5, 1, 1)?